### ISSUE TITLE
Display current component and application in shell prompt.

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#488]
Document configuration steps and create configuration script that will configure bash and zsh to show current component and application in the command prompt.

Most of the people configure they shell to show current git branch. It would be nice to have something similar for odo components.

**Comment by cdrage** [comment]
@kadel 

I did some minor research as well as started integrating this.

I have a question however! At the moment, `odo component get` requires an OpenShift cluster to be active. Unfortunately, when implementing this..

Example:

```sh
__odo_ps1()
{
    # Get current context
    CONTEXT=$(odo component get)

    if [ -n "$CONTEXT" ]; then
        echo "(odo: ${CONTEXT})"
    fi
}
```

It ends up adding a 1-2 second delay (as internally, we check and grep OpenShift through its API).

Is there an alternative way of implementing this such as grepping `~/.kube/odo` (although at the moment, it only lists active applications but not what the currently active component is...)

Context (and source): https://pracucci.com/display-the-current-kubelet-context-in-the-bash-prompt.html

**Comment by kadel** [comment]
Would it make sense to add a flag to `odo component get` that will skip API check to make it fast?
I would like to have login in one place. Because if we change the config file structure in future (or we move this information somewhere else), we will have to go and update this bash script also. I can guarantee that we will forget about it. :-)

**Comment by cdrage** [comment]
@kadel I added an edit two days ago, but I ended up grepping ~/.kube/odo instead and it works well! :+1:

**Comment by kadel** [comment]
I don't have doubts about it not working, more about future maintenance issues. :-( 
Parsing yaml file using grep doesn't sound right to me.

**Comment by cdrage** [comment]
@kadel Yeah, it would be a lot easier doing it via `odo component get` rather than relying on the configuration infrastructure.

Sure, the code is here:

```sh
__odo_ps1()
{

    # Get current context
    COMPONENT=$(cat ~/.kube/odo | grep -A 3 "active: true" | grep "activeComponent:" | sed "s/  activeComponent: //")
    APPLICATION=$(cat ~/.kube/odo | grep -A 3 "active: true" | grep "name:" | sed "s/  name: //")

    if [ -n "$COMPONENT" ] || [ -n "$APPLICATION" ]; then
        echo "odo: ${APPLICATION} - ${COMPONENT}"
    fi
}
```

**Comment by cdrage** [comment]
Then you can plug it into PS1 via:

```sh
export PS1="$(__odo_ps1) \$"
```

I was thinking of adding this as a parameter to `odo completion`. For example: `odo completion --terminal` or something similar that will set the PS1 automatically.

**Comment by kadel** [comment]
> @kadel Yeah, it would be a lot easier doing it via odo component get rather than relying on the configuration infrastructure.

It should be that hard to skip connection test that we do every time new occlient is initialized.
It could be just one parameter to `occlient.New` something like `connectionCheck`.
Then we could have `odo component get --no-cluster-check` or `--no-conn-check` or something similar and use it when setting $PS1

**Comment by cdrage** [comment]
Once https://github.com/redhat-developer/odo/pull/529 is merged, the work will continue on PS1 / terminal status script.

**Comment by cdrage** [comment]
Closed via: https://github.com/redhat-developer/odo/pull/536
